plugins/modules/luks_device.py

```diff
@@ -523,6 +572,27 @@ def is_luks(self, device):
         result = self._run_command([self._cryptsetup_bin, 'isLuks', device])
         return result[RETURN_CODE] == 0
 
+    def get_luks_type(self, device):
+        ''' get the luks type of a device
+        '''
+        if self.is_luks(device):
+            result = self._run_command([self._cryptsetup_bin, 'luksDump', device])
+            for line in result[STDOUT].splitlines():
+                if 'Version' in line:
+                    version = line.split()[1]
+                    return 'luks2' if version == '2' else 'luks1'
+        return None
+
+    def is_luks_slot_set(self, device, keyslot):
+        ''' check if a keyslot is set
+        '''
+        luks_header = result = self._run_command([self._cryptsetup_bin, 'luksDump', device])
+        if result[RETURN_CODE] != 0:
+            raise ValueError('Error while dumping LUKS header from %s' % (device, ))
+        result_luks1 = 'Key Slot %d: ENABLED' % (keyslot) in luks_header[STDOUT]
+        result_luks2 = '%d: luks2' % (keyslot) in luks_header[STDOUT]
```

Comment:
If `keyslot == 1`, this will also match `11: luks2`. A regular expression (or another character before `%d`) should help.